### ISSUE TITLE
Empty <html> lang attributes report as fail of html-has-lang, pass for html-lang-valid

### ISSUE THREAD

**Issue by thibaudcolas** [dequelabs/axe-core#3624]
### Product

axe-core

### Product Version

4.4.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

For `<html lang="">`, `html-has-lang` fails, and `html-lang-valid` passes. This is a similar issue to #3623, where I would expect the rules to be consistent with one-another.

#### Expectation

Both rules should be consistent. If `lang=""` is indeed a valid lang attribute value, then `html-has-lang` should pass (both rules should pass). If it’s not a valid value, then either both rules should fail, or only `html-lang-valid` should fail.

#### Actual

Currently, `<html lang="">` passes `html-lang-valid`.

#### How to Reproduce

Here are two sites where this happens, and Axe CLI commands:

```bash
axe -r html-has-lang,html-lang-valid https://www.koldijkbv.nl/ --save empty-lang.json
axe -r html-has-lang,html-lang-valid https://loyalty.posnania.eu/ --save empty-lang.json
```

In each case, see the `html-has-lang` violation while `html-lang-valid` is marked as a "pass".

#### Additional context

Similarly to #3623, I was also tempted to report this to Lighthouse instead / in addition to Axe, as its overly simplified audit messages make this even more confusing. It would report:

- An error "`<html> element does not have a [lang] attribute`"
- And a pass "`<html> element has a valid value for its [lang] attribute`"

**Comment by WilcoFiers** [comment]
We try to avoid having axe-core report the same problem more than once. There's not much more to it than that. It's important to know that even if a rule passes, it's always possible for there to be some other issue.

We could update the rule selector of `html-lang-valid` to skip empty lang attributes. That way the rule would be inapplicable, rather than passed. Would that have helped with your concern?

**Comment by thibaudcolas** [comment]
Hi @WilcoFiers, thank you for looking into this!

Researching this further, I learned that the [HTML page has lang attribute ACT rule](https://act-rules.github.io/rules/b5c3f8) specifically expects a failure for `<html lang="">`, so I’m hesitant to suggest any changes to Axe’s `html-has-lang`’s . Looking at [HTML page lang attribute has valid language tag](https://act-rules.github.io/rules/bf051a), its applicability says:

> […] has a lang attribute that is neither empty ("") nor only ASCII whitespace […]

… so your suggestion to have the rule marked as inapplicable in this case seems spot on.

Unfortunately although this scenario is documented in the ACT rule description, there isn’t a test case for it there, so implementations seem to differ (Axe passes, [Alfa seems to report as inapplicable](https://github.com/Siteimprove/alfa/blob/main/packages/alfa-rules/src/sia-r5/rule.ts#L29)).

**Comment by thibaudcolas** [comment]
Cool! 

> I'll leave this ticket open for us to update this selector. It should be a fairly straightforward tweak to the rules elector.

Would this be something along the lines of `html[lang]:not([lang=""]), html[xml\\:lang]:not([xml\\:lang=""])` in place of [html-lang-valid’s existing `html[lang], html[xml\\:lang]`](https://github.com/dequelabs/axe-core/blob/2fb662f594073dc68be2891e8341ef1b5bece042/lib/rules/html-lang-valid.json)? If so I’ll happily give this a shot if it helps.

**Comment by WilcoFiers** [comment]
@thibaudcolas Pretty much, yes. We'll need some integration tests to go with it in `test/integration/full/html-lang-valid/`. A PR would be much appreciated!

**Comment by thibaudcolas** [comment]
I gave this a go at #3663 but unfortunately got stuck with an issue with the CSS selector processing.

**Comment by padmavemulapati** [comment]
Validated with the latest code base,
for the test snippet ` <html lang="">` , `html-has-lang` failing and as well `valid-lang` is in `inapplicable`



![Image](https://user-images.githubusercontent.com/47409579/205640982-cb3c3702-1cab-464d-98b9-3387ec9fba90.png)



![Image](https://user-images.githubusercontent.com/47409579/205641214-8b4f91f9-e286-43c9-a91f-77231d8d7ec5.png)
